devices/Scd30/Scd30Sensor.cs

```diff
@@ -0,0 +1,419 @@
+            return Temperature.FromDegreesCelsius(BinaryPrimitives.ReadUInt16BigEndian(data) / 100d);
+        }
+
+        /// <summary>
+        /// <para>
+        ///     Measurements of CO2 concentration based on the NDIR principle are influenced by altitude. SCD30 offers to compensate
+        ///     deviations due to altitude by using this command. Setting altitude is disregarded when an ambient pressure is given to
+        ///     the sensor, please see section 1.4.1.
+        /// </para>
+        /// <para>
+        ///     Altitude value is saved in non-volatile memory. The last set value will be used for altitude compensation after repowering.
+        /// </para>
+        /// </summary>
+        /// <param name="altitudeAboveSeaLevel">Height over sea level above 0.</param>
+        public void SetAltitudeCompensation(Length altitudeAboveSeaLevel)
+        {
+            ModbusWriteSingleHoldingRegister(0x38, (ushort)altitudeAboveSeaLevel.Meters);
+        }
+
+        /// <summary>
+        /// Get the currently configured altitude compensation. See <see cref="SetAltitudeCompensation(Length)"/> for more information.
+        /// </summary>
+        /// <returns>The currently configured altitude compensation.</returns>
+        public Length GetAltitudeCompensation()
+        {
+            var data = ModbusReadHoldingRegisters(0x38, 1);
+            return Length.FromMeters(BinaryPrimitives.ReadUInt16BigEndian(data));
+        }
+
+        /// <summary>
+        /// This command can be used to read out the firmware version of SCD30 module.
+        /// </summary>
+        /// <returns>The firmware version of the SCD30 sensor.</returns>
+        public Version ReadFirmwareVersion()
+        {
+            var data = ModbusReadHoldingRegisters(0x20, 1);
+            return new Version(data[0], data[1]);
+        }
+
+        /// <summary>
+        /// <para>
+        ///     The SCD30 provides a soft reset mechanism that forces the sensor into the same state as after powering up without the need
+        ///     for removing the power-supply. It does so by restarting its system controller. After soft reset the sensor will reload all calibrated
+        ///     data. However, it is worth noting that the sensor reloads calibration data prior to every measurement by default. This includes
+        ///     previously set reference values from ASC or FRC as well as temperature offset values last setting.
+        /// </para>
+        /// <para>
+        ///     The sensor is able to receive the command at any time, regardless of its internal state.
+        /// </para>
+        /// </summary>
+        public void SoftReset()
+        {
+            ModbusWriteSingleHoldingRegister(0x34, 1);
+        }
+
+        private void ModbusWriteSingleHoldingRegister(ushort register, ushort value)
+        {
+            // Function 0x6 request is always 8 bytes
+            var request = new byte[8];
```

Comment:
I can't. I do need the byte[] because `SerialPort.Write(..)` does not work with `SpanByte`. I will refactor to create a SpanByte directly for Slice manipulation (which happens a few times) in the function.

Reply:
You can directly do that:

```csharp
SpanByte requestSpan = new byte[8];
// keep all the rest 
// Execute
ModbusWriteAndRead(requestSpan.ToArray(), response);
```

(and just rename the requestSpan to request to get closer to your code).
That may be as well easier to read.

Reply:
True, but your original comment rightfully stated the advantage of using SpanByte is to avoid duplicating the byte array. Introducing the ToArray(), which does instantiate a new byte array, just for readability, seems unnecessary.

We're really just nitpicking here though. The array is only 8 bytes ;-)